# Case Study 
## The annual Percentage Rate of Interest Charged on New-Car Loans at Nine of the 					Largest Banks in Six American Cities is provided and It is believed that the mean interest rate is same across the six american cities.  Conduct the hypothesis test and provide your inference


Use level of significance $\alpha$ = 0.05

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from statsmodels.stats.anova import anova_lm, _get_covariance
from statsmodels.formula.api import ols

In [21]:
df = pd.read_csv('car_insurance_rate.csv', delimiter="\t")

In [22]:
df.head()

Atlanta  Chicago  Houston  Memphis  New York  Philadelphia
0    13.75    14.25    14.00    15.00     14.50         13.50
1    13.75    13.00    14.00    14.00     14.00         12.25
2    13.50    12.75    13.51    13.75     14.00         12.25
3    13.50    12.50    13.50    13.59     13.90         12.00
4    13.00    12.50    13.50    13.25     13.75         12.00

In [23]:
df.rename(columns = {'New York':'NewYork'}, inplace = True)

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9 entries, 0 to 8
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Atlanta       9 non-null      float64
 1   Chicago       9 non-null      float64
 2   Houston       9 non-null      float64
 3   Memphis       9 non-null      float64
 4   NewYork       9 non-null      float64
 5   Philadelphia  9 non-null      float64
dtypes: float64(6)
memory usage: 560.0 bytes


In [27]:
df.Atlanta.unique()

array([13.75, 13.5 , 13.  , 12.75, 12.5 ])

In [28]:
df.Chicago.unique()

array([14.25, 13.  , 12.75, 12.5 , 12.4 , 12.3 , 11.9 ])

In [29]:
df.Houston.unique()

array([14.  , 13.51, 13.5 , 13.25, 13.  , 12.5 ])

In [30]:
df.Memphis.unique()

array([15.  , 14.  , 13.75, 13.59, 13.25, 12.97, 12.5 , 12.25, 11.89])

In [31]:
df['NewYork'].unique()

array([14.5 , 14.  , 13.9 , 13.75, 13.25, 13.  , 12.5 , 12.45])

In [32]:
df.Philadelphia.unique()

array([13.5 , 12.25, 12.  , 11.9 ])

In [33]:
df.describe()

Atlanta    Chicago    Houston    Memphis    NewYork  Philadelphia
count   9.000000   9.000000   9.000000   9.000000   9.000000      9.000000
mean   13.194444  12.611111  13.306667  13.244444  13.483333     12.200000
std     0.446825   0.710096   0.556350   0.967007   0.718940      0.504356
min    12.500000  11.900000  12.500000  11.890000  12.450000     11.900000
25%    13.000000  12.300000  13.000000  12.500000  13.000000     12.000000
50%    13.000000  12.500000  13.500000  13.250000  13.750000     12.000000
75%    13.500000  12.750000  13.510000  13.750000  14.000000     12.250000
max    13.750000  14.250000  14.000000  15.000000  14.500000     13.500000

In [36]:
df_melt=df.melt(var_name='city') # using melt will reshape the dataframe and make it suitable for ols model

In [37]:
df_melt.head(10)

city  value
0  Atlanta  13.75
1  Atlanta  13.75
2  Atlanta  13.50
3  Atlanta  13.50
4  Atlanta  13.00
5  Atlanta  13.00
6  Atlanta  13.00
7  Atlanta  12.75
8  Atlanta  12.50
9  Chicago  14.25

# 1. Postulate the Null and Alternate Hypothesis

## The Hypothesis for the One Way ANOVA are:
    
## $H_0$: $Mu_A$ = $Mu_C$ = $Mu_H$ = $Mu_M$ = $Mu_N$ = $Mu_P$. That is mean interest rate in all siz American state is equal
## $H_a$: Atleast one of $Mu_A$, $Mu_C$, $Mu_H$, $Mu_M$, $Mu_N$, and $Mu_P$ is different. Atleast one of the 6 state has different mean interest rate

In [39]:
#Degree of freedom between groups (6 states)
no_of_states = df.shape[1]
degree_of_freedom_between_groups= no_of_states-1
print('Degree of Freedom between groups',degree_of_freedom_between_groups)

Degree of Freedom between groups 5


In [42]:
#Degree of freedom within each groups (states)
no_of_observation_for_each_group=df.shape[0]
total_no_of_observations = no_of_observation_for_each_group * no_of_states
degree_of_freedom_within_each_groups = total_no_of_observations - no_of_states
print('Degree of Freedom within groups',degree_of_freedom_within_each_groups)

Degree of Freedom within groups 48


# 2. Perform the analysis of variances test on the dataset

In [38]:
formula = 'value ~ city'
model = ols(formula, df_melt).fit()
anova_table = anova_lm(model)
pd.set_option('expand_frame_repr', False)
print(anova_table)

            df     sum_sq   mean_sq         F    PR(>F)
city       5.0  10.945667  2.189133  4.829385  0.001175
Residual  48.0  21.758133  0.453294       NaN       NaN


# 3. What are sum of squared variance within and sum of squared variance between?

`Sum of Squared variance Within (SSW or SSE)`  is the sum over the sums of squared deviations of scores around their group’s mean


`Sum of Squared variance Between (SSB)` is the sum of squared deviation of the group means from the grand mean.

In [49]:
SSB = anova_table['sum_sq'][0]  ## Also known as SSBetween
SSW = anova_table['sum_sq'][1]  ## Also known as SSError
print('Sum of squared variance within Groups (SSW or SSE) is -',SSW ,'\nSum of squared variance Between groups (SSB) is -',SSB)

Sum of squared variance within Groups (SSW or SSE) is - 21.758133333333333 
Sum of squared variance Between groups (SSB) is - 10.945666666666636


# 4. Calculate sum of squared total for above

${SST = SSB + SSW}$

In [50]:
SST = SSB + SSW
print("Sum of Squared total is -",SST)

Sum of Squared total is - 32.70379999999997


# 5. What is the significance of F dist/statistic?

`F-Distribution or F-Statistic is the ratio of MSB to MSW. `

It gives the degree of how relatively greater the difference is `between group means` (MSB) compared to `within group variance` (MSW)

If the ratio is greater than expected will mean that not all the group means are the same and at least one mean is substantially different.

# 6. Interpret the results and present the conclusion

In [51]:
anova_table['PR(>F)'][0]

0.001174551414504068

`p-value is 0.001175`, which is less than level of significance $\alpha$ 0.05.

`This implies that there is significant evidence to reject the null hypothesis`. 
`It means that the mean interest rate of each of the six city is not same and atleast one of the city has different interest rate`.